script_parameters.yml

```diff
@@ -0,0 +1,238 @@
+          my_file_url = os.environ['MY_FILE_URL']
+
+          my_custom_credential = (os.environ['MY_CUSTOM_CREDENTIAL_USERNAME'], os.environ['MY_CUSTOM_CREDENTIAL_PASSWORD'])
+          my_aws_credential = (os.environ['MY_AWS_CREDENTIAL_ACCESS_KEY_ID'], os.environ['MY_AWS_CREDENTIAL_SECRET_ACCESS_KEY'])
+          my_db_credential = (os.environ['MY_DB_CREDENTIAL_USERNAME'], os.environ['MY_DB_CREDENTIAL_PASSWORD'])
+      on-success:
+        - r_script
+
+    r_script:
+      action: civis.scripts.r
+      input:
+        required_resources:
+          cpu: 256
+          memory: 512
+          disk_space: 1
+        params:
+          - name: my_integer
+            type: integer
+          - name: my_float
+            type: float
+          - name: my_string
+            type: string
+          - name: my_file
+            type: file
+          - name: my_custom_credential
+            type: credential_custom
+          - name: my_aws_credential
+            type: credential_aws
+          - name: my_db_credential
+            type: credential_redshift
+        arguments:
+          my_integer: 1234
+          my_float: 0.1234
+          my_string: this is a string
+          my_file: 0               # ID of the file
+          my_custom_credential: 0  # ID of the credential
+          my_aws_credential: 0
+          my_db_credential: 0
+        source: |
+          # Parameter values are added to your container as environment variables.
+          # See the script params reference for a full list.
+
+          # Environment variables are retrieved as strings, so if you want to use
+          # them as numbers, you'll need to do type conversion.
+          my_integer <- as.numeric(Sys.getenv('MY_INTEGER'))
+          my_float <- as.numeric(Sys.getenv('MY_FLOAT'))
+
+          my_string <- Sys.getenv('MY_STRING')
+          my_file_url <- Sys.getenv('MY_FILE_URL')
+
+          my_custom_credential_username <- Sys.getenv('MY_CUSTOM_CREDENTIAL_USERNAME')
+          my_custom_credential_password <- Sys.getenv('MY_CUSTOM_CREDENTIAL_PASSWORD')
+          my_aws_credential_access_key_id <- Sys.getenv('MY_AWS_CREDENTIAL_ACCESS_KEY_ID')
+          my_aws_credential_secret_access_key <- Sys.getenv('MY_AWS_CREDENTIAL_SECRET_ACCESS_KEY')
+          my_db_credential_username <- Sys.getenv('MY_DB_CREDENTIAL_USERNAME')
+          my_db_credential_password <- Sys.getenv('MY_DB_CREDENTIAL_PASSWORD')
+      on-success:
+        - custom_script
+
+    custom_script:
+      action: civis.scripts.custom
+      input:
+        # Custom scripts are built off of templates, which already define the
+        # `params` list, so all you need to provide is the `arguments` hash.
+        # For this example we'll pretend that the template has a string parameter
+        # called `my_string`.
+        from_template_id: 0  # ID of the template backing your custom script
+        arguments:
+          my_string: this is a string
```

Comment:
might want to add the 'on-success' here too for consistency

Reply:
I think that'd be good if I had another job to go to after this one, but this is the last one in the chain.